pkg/source/file/source.go

```diff
@@ -116,6 +117,7 @@ func (s *Source) Init(context api.Context) error {
 		SourceName:   s.name,
 		Level:        IsolationLevel(s.config.Isolation),
 	}
+	globalLineEnd.AddLineEnd(s.pipelineName, s.name, &s.config.LineEnd)
```

Comment:
Called in the `component.Init()` phase, because the config may not be initialized at this time